### PR TITLE
more principled peer resoluation

### LINKED ISSUE #1257: refine `Peer` / `PeerCandidate` types and implement SRV record support
Consensus protocol version 15 differs from our currently implemented v14 only in advertising the ability to use SRV records in peer discovery. Properly implementing this should first clean up the internal terminology so that we will arrive at the following state:

- `ConnectionId` stays as the only unambiguous means of referring to a particular bearer
- `Peer` shall be the directly addressable network address of a peer, i.e. the dialling address when constructing a bearer or the remote address after having accepted an inbound bearer
- `PeerCandidate` shall be the rich set of peer discovery outcomes that can be resolved into a `Peer` before attempting to construct a bearer

For now, a `Peer` can therefore be either an IPv4 or IPv6 address paired with a socket number. A `PeerCandidate` can be either a `Peer` or a DNS name paired with a socket number or a DNS name referring to SRV records. When selecting a `PeerCandidate` for a new outgoing bearer, it must first be resolved to a `Peer` within the `peer_selection` stage and then handed to the network `Manager`. `PeerCandidate` is thus only used by `peer_selection`, `peer_sharing`, the ledger’s `PoolParams::relays`, and the CLI/TUI.

As a consequence, the `peer_selection` / Performance resource will have to maintain a mapping from `Peer` back to the `PeerCandidate` it was resolved from (if any) in order to attribute all metrics and behaviours to the same entity. It will also need to treat inbound peers whose socket address cannot be connected to as non-candidates — inbound peers can only enter the candidate pool after an outbound connection has been tried and has succeeded.

### PR DESCRIPTION
**Pull request by rkuhn**
fixes #1257

This PR ensures a proper separation between PeerCandidate, which can be a name that must be
resolved, covering all CIP0155 features, and a Peer that is a network-level diallable address
(currently isomorphic to SocketAddr, but kept extensible).

- **introduce PeerCandidate and make `Peer: Clone`**
- **add eff.detach() to pure-stage**
- **make `Peer: Copy` and split off PeerCandidate**
- **implement support for SRV records**
- **display PeerCandidate in TUI**
- **resolve directly before dialling**

## Summary by CodeRabbit

- **New Features**
  - Added support for hostname- and SRV-based peer addresses with DNS resolution.
  - Peer snapshots preserve unresolved relay names for later resolution.
  - The peer-selection view shows bootstrap names alongside resolved addresses.
  - Added detached external effects that continue asynchronously without blocking stage progress.
  - Connections now dial resolved peer addresses.

- **Bug Fixes**
  - Improved handling and reporting of invalid or failed peer resolution.
  - Prevented bootstrap database caches for different epochs from being mixed.

- **Documentation**
  - Updated relay configuration and peer-resolution trace documentation.

### CODE REVIEW

File: crates/amaru-consensus/src/effects/resolve.rs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+async fn resolve_candidate(candidate: &PeerCandidate) -> Option<Peer> {
+    #[cfg(all(not(target_family = "wasm"), not(target_arch = "riscv32")))]
+    {
+        amaru_network::resolve::resolve_peer_candidate(candidate).await
+    }
+    #[cfg(any(target_family = "wasm", target_arch = "riscv32"))]
+    {
+        let _ = candidate;
+        None
+    }
```
[the remark this answers is not in the document]
Comment (rkuhn) [reply crates/amaru-consensus/src/effects/resolve.rs:70]: there’s no networking for wasm anyway

File: crates/amaru-tui/src/model/telemetry_update.rs [hunk 2]
```diff
@@ -242,6 +243,18 @@ impl Model {
         peer.mark_connected(record);
     }
 
+    fn update_peer_resolved(&mut self, record: &TelemetryRecord) {
+        let address = protocols::peer_selection::peer::RESOLVED::peer(record);
+        let candidate = protocols::peer_selection::peer::RESOLVED::candidate(record);
+        if candidate == address {
+            return;
+        }
+        self.resolved_candidates.insert(address.to_string(), candidate.to_string());
```
[the remark this answers is not in the document]
Comment (rkuhn) [reply crates/amaru-tui/src/model/telemetry_update.rs:252]: This is unlikely to be a problem: not many resolutions per minute.
